ktor-utils/common/src/io/ktor/util/Attributes.kt

```diff
@@ -10,7 +10,26 @@ package io.ktor.util
  * @param name is a name of the attribute for diagnostic purposes
  */
 public class AttributeKey<T>(public val name: String) {
-    override fun toString(): String = if (name.isEmpty()) super.toString() else "AttributeKey: $name"
+
+    override fun toString(): String = when {
+        name.isEmpty() -> super.toString()
+        else -> "AttributeKey: $name"
+    }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
I'am not sure about this: this breaks semantics since different attribute keys with accidentally same names may start colliding. Imagine several unrelated people using different attributes with silly names like `"attribute1"` may surprisingly clash each other

Reply:
thanks, fixed

Reply:
Please, resolve obsolete conversations if possible :)